test/services/company.appointments.service.unit.ts

```diff
@@ -91,5 +91,42 @@ describe("Company Appointments service test", () => {
           expect(createAndLogError).toHaveBeenCalledWith(expect.stringContaining(`${APPOINTMENT_ID}`));
         });
     });
+
+    it("Should obfuscate residential address when isSecureOfficer is true", async () => {
+      const companyResource = clone(validCompanyAppointmentResource);
+      companyResource.resource.isSecureOfficer = true;
+      mockGetCompanyAppointmentFullRecord.mockResolvedValueOnce(companyResource);
+
+      const returnedAppointment: CompanyAppointment = await getCompanyAppointmentFullRecord(session, COMPANY_NUMBER, APPOINTMENT_ID);
+
+      expect(returnedAppointment.usualResidentialAddress.addressLine1).toEqual("********");
+      expect(returnedAppointment.usualResidentialAddress.addressLine2).toEqual("********");
+      expect(returnedAppointment.usualResidentialAddress.careOf).toEqual("********");
+      expect(returnedAppointment.usualResidentialAddress.country).toEqual("********");
+      expect(returnedAppointment.usualResidentialAddress.locality).toEqual("********");
+      expect(returnedAppointment.usualResidentialAddress.poBox).toEqual("********");
+      expect(returnedAppointment.usualResidentialAddress.postalCode).toEqual("********");
+      expect(returnedAppointment.usualResidentialAddress.premises).toEqual("********");
+      expect(returnedAppointment.usualResidentialAddress.region).toEqual("********");
+    });
+
+    test.each([false, null])
```

Comment:
regarding your comment
`Sonar says I am missing a test scenario but I'm not sure I am. Maybe someone can see the scenario that's missing?`

try adding `undefined` in `test.each[]` array, if I am not wrong the null and undefined are slightly different.

Reply:
Didn't help unfortunately